packages/typespec-vscode/src/npm-utils.ts

```diff
@@ -0,0 +1,143 @@
+import fs from "fs";
+import path from "path";
+import semver from "semver";
+import logger from "./log/logger.js";
+import { ExecOutput, loadModule, spawnExecution, spawnExecutionEvents } from "./utils.js";
+
+export enum InstallationAction {
+  Install = "Install",
+  Upgrade = "Upgrade",
+  Skip = "Skip",
+  Cancel = "Cancel",
+}
+
+export enum npmDependencyType {
+  dependencies = "dependencies",
+  peerDependencies = "peerDependencies",
+  devDependencies = "devDependencies",
+}
+
+export interface NpmPackageInfo {
+  name: string;
+  version?: string;
+  resolved?: string;
+  overridden?: string;
+  dependencies?: Record<string, NpmPackageInfo>;
+}
+
+export class NpmUtil {
+  private cwd: string;
+
+  constructor(cwd: string) {
+    this.cwd = cwd;
+  }
+
+  public async npmInstallPackages(
+    packages: string[] = [],
+    options: any = {},
+    on?: spawnExecutionEvents,
+  ): Promise<ExecOutput> {
+    return spawnExecution("npm", ["install", ...packages], this.cwd, on);
+  }
+
+  public async ensureNpmPackageInstall(
+    packageName: string,
+    version?: string,
+  ): Promise<{ action: InstallationAction; version?: string }> {
```

Comment:
the name seems not match the implementation...

Reply:
This function will calculate the install action (install, skip, cancel). change the function name to `ensureNpmPackageInstallAction` to make it clear.

Reply:
The name is still confusing to me. @timotheeguerin , @chrisradek to see whether they have better idea for it.